crates/precompile/src/kzg_point_evaluation.rs

```diff
@@ -0,0 +1,108 @@
+use crate::{Error, Precompile, PrecompileAddress, PrecompileResult, B160};
+use c_kzg::{Bytes32, Bytes48, KzgProof, KzgSettings};
+use revm_primitives::{hex_literal::hex, Env};
+use sha2::{Digest, Sha256};
+
+pub const POINT_EVALUATION: PrecompileAddress = PrecompileAddress(ADDRESS, Precompile::Env(run));
+
+const ADDRESS: B160 = crate::u64_to_b160(0x0A);
+const GAS_COST: u64 = 50_000;
+const VERSIONED_HASH_VERSION_KZG: u8 = 0x01;
+
+/// `U256(FIELD_ELEMENTS_PER_BLOB).to_be_bytes() ++ BLS_MODULUS.to_bytes32()`
+const RETURN_VALUE: &[u8; 64] = &hex!(
+    "0000000000000000000000000000000000000000000000000000000000001000"
+    "73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001"
+);
+
+fn run(input: &[u8], gas_limit: u64, env: &Env) -> PrecompileResult {
+    if gas_limit < GAS_COST {
+        return Err(Error::OutOfGas);
+    }
+    if input.len() != 192 {
+        return Err(Error::BlobInvalidInputLength);
+    }
+
+    // Verify commitment matches versioned_hash
+    let commitment = &input[96..144];
+    let versioned_hash = &input[0..32];
```

Comment:
can we swap the order of these two vars, so we start slicing from the beginning?